internal/app/init/internal/platform/iso/iso.go

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package iso
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/talos-systems/talos/internal/pkg/install"
+	"github.com/talos-systems/talos/pkg/crypto/x509"
+	"github.com/talos-systems/talos/pkg/userdata"
+	"golang.org/x/sys/unix"
+)
+
+// ISO is a platform for installing Talos via an ISO image.
+type ISO struct{}
+
+// Name implements the platform.Platform interface.
+func (i *ISO) Name() string {
+	return "ISO"
+}
+
+// UserData implements the platform.Platform interface.
+func (i *ISO) UserData() (data *userdata.UserData, err error) {
+	data = &userdata.UserData{
+		Security: &userdata.Security{
+			OS: &userdata.OSSecurity{
+				CA:       &x509.PEMEncodedCertificateAndKey{},
+				Identity: &x509.PEMEncodedCertificateAndKey{},
+			},
+			Kubernetes: &userdata.KubernetesSecurity{
+				CA: &x509.PEMEncodedCertificateAndKey{},
+			},
+		},
+		Install: &userdata.Install{
+			Force: true,
+			Boot: &userdata.BootDevice{
+				InstallDevice: userdata.InstallDevice{
+					Device: "/dev/sda",
+					Size:   512 * 1000 * 1000,
+				},
+			},
+			Root: &userdata.RootDevice{
+				InstallDevice: userdata.InstallDevice{
+					Device: "/dev/sda",
+					Size:   2048 * 1000 * 1000,
+				},
+			},
+			Data: &userdata.InstallDevice{
+				Device: "/dev/sda",
+				Size:   2048 * 1000 * 1000,
+			},
+			Wipe: true,
+		},
+	}
+
+	return data, nil
+}
+
+// Prepare implements the platform.Platform interface.
+func (i *ISO) Prepare(data *userdata.UserData) (err error) {
+	return install.Prepare(data)
+}
+
+// Install implements the platform.Platform interface.
+func (i *ISO) Install(data *userdata.UserData) error {
+	// Prompt the user for the base URL of all assets.
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Print("Enter the base asset URL:")
```

Comment:
Do we want to do this? I'm expecting all of this would already be contained in the iso.

Reply:
Ahh actually good point, never thought of that. Maybe we make it optional and default to what is in the ISO? 🤔 maybe not though... kinda makes sense to bundle everything together, it ensures no compatibility issues.